dlt/sources/sql_database/arrow_helpers.py

```diff
@@ -20,5 +19,10 @@ def row_tuples_to_arrow(
     from dlt.common.libs.pyarrow import row_tuples_to_arrow as _row_tuples_to_arrow
 
     return _row_tuples_to_arrow(
-        rows, caps or DestinationCapabilitiesContext.generic_capabilities(), columns, tz
+        rows=rows,
+        caps=Container().get(DestinationCapabilitiesContext)
+        or DestinationCapabilitiesContext.generic_capabilities(),
+        columns=columns,
+        tz=tz,
+        safe_arrow_conversion=True,
```

Comment:
OK! but if we keep with_config, `safe_arrow_conversion` (if present in the signature) would be configurable. just FYI. we can come back to that later